services/export/shared/transactionsExport.js

```diff
@@ -0,0 +1,462 @@
+			);
+			allTransactions.push(...transactions);
+
+			const incomingCrossChainTransferTxs = await getCrossChainTransferTransactionInfo({
+				...params,
+				timestamp,
+			});
+
+			if (incomingCrossChainTransferTxs.length) {
+				allTransactions.push(...incomingCrossChainTransferTxs);
+			}
+
+			if (day !== getToday()) {
+				if (transactions.length) {
+					partials.write(partialFilename, JSON.stringify(transactions));
+				} else {
+					// Flag to prevent unnecessary calls to core/storage space usage on the file cache
+					const RETENTION_PERIOD = getDaysInMilliseconds(config.cache.partials.retentionInDays);
+					await noTransactionsCache.set(partialFilename, true, RETENTION_PERIOD);
+				}
+			}
+		}
+		/* eslint-enable no-await-in-loop */
+	}
+
+	const excelFilename = await getExcelFilenameFromParams(params);
+	const currentChainID = await getCurrentChainID();
+
+	// Add duplicate entry with zero fees for self token transfer transactions
+	allTransactions.forEach((tx, i, arr) => {
+		if (checkIfSelfTokenTransfer(tx) && !tx.isSelfTokenTransferCredit) {
+			arr.splice(i + 1, 0, { ...tx, fee: '0', isSelfTokenTransferCredit: true });
+		}
+	});
+
+	const normalizedTransactions = await Promise.all(allTransactions
+		.map(async (t) => normalizeTransaction(getAddressFromParams(params), t, currentChainID)));
+	const metadata = await getMetadata(params, currentChainID);
+
+	const workBook = new excelJS.Workbook();
+	const transactionExportSheet = workBook.addWorksheet(config.excel.sheets.TRANSACTION_HISTORY);
+	const metadataSheet = workBook.addWorksheet(config.excel.sheets.METADATA);
+	transactionExportSheet.columns = fields.transactionMappings;
+	metadataSheet.columns = fields.metadataMappings;
+
+	transactionExportSheet.addRows(normalizedTransactions);
+	metadataSheet.addRows([metadata]);
+
+	await workBook.xlsx.writeFile(`${config.cache.exports.dirPath}/${excelFilename}`);
+};
+
+const scheduleTransactionExportQueue = Queue(
+	config.endpoints.redis,
+	config.queue.scheduleTransactionExport.name,
+	exportTransactions,
+	config.queue.scheduleTransactionExport.concurrency,
+);
+
+const scheduleTransactionHistoryExport = async (params) => {
+	// Schedule only when index is completely built
+	const isBlockchainIndexReady = await requestIndexer('isBlockchainFullyIndexed');
+	if (!isBlockchainIndexReady) throw new ValidationException('Blockchain index is not yet ready.');
+
+	const exportResponse = {
+		data: {},
+		meta: {
+			ready: false,
+		},
+	};
+
+	const { publicKey } = params;
+	const address = getAddressFromParams(params);
+
+	// Validate if account exists
+	const isAccountExists = await validateIfAccountExists(address);
+	if (!isAccountExists) throw new NotFoundException(`Account ${address} not found.`);
+
+	// Validate if account has transactions
+	const isAccountHasTransactions = await validateIfAccountHasTransactions(address);
+	if (!isAccountHasTransactions) throw new NotFoundException(`Account ${address} has no transactions.`);
```

Comment:
An account may not have any transactions but if it already has some balance, we can still return only the metadata. Let us please add the same logic to the validation above.